freud/order.pyx

```diff
@@ -598,41 +605,59 @@ cdef class Steinhardt(_PairCompute):
     @property
     def l(self):  # noqa: E743
         """unsigned int: Spherical harmonic quantum number l."""
-        return self.thisptr.getL()
+        ls = self.thisptr.getL()
+        # list conversion is necessarily explicit as otherwise CI Cython
+        # complains about compiling the expression with two different types.
+        return ls[0] if ls.size() == 1 else list(ls)
```

Comment:
Or just dodge the Cython confusion entirely by pre-emptively creating a list. Also applies below.
```suggestion
        ls = list(self.thisptr.getL())
        return ls[0] if len(ls) == 1 else ls
```